internal/ips/request.go

```diff
@@ -46,35 +34,80 @@ func (c *Client) Request() *cobra.Command {
 		Short: "Request a block of IP addresses.",
 		Long:  "Requests either a block of public IPv4 addresses or global IPv4 addresses for your project in a specific metro or facility.",
 		Example: `  # Requests a block of 4 public IPv4 addresses in Dallas:
-  metal ip request -p $METAL_PROJECT_ID -t public_ipv4 -q 4 -m da`,
+  metal ip request -p $METAL_PROJECT_ID -t public_ipv4 -q 4 -m da
+
+  metal ips request -v <vrfid> -t vrf -i <cidr, 22-29> -n <network, IP address> --tags foo --tags bar -u '{"my":"goodness"}' --details "i don't think VRF users need this or will see it after submitting the request"'`,
 
 		RunE: func(cmd *cobra.Command, args []string) error {
+			var (
+				req                         *metal.IPReservationRequestInput
+				vrfReq                      *metal.VrfIpReservationCreateInput
+				requestIPReservationRequest *metal.RequestIPReservationRequest
+			)
 			cmd.SilenceUsage = true
+			typeFlag, _ := cmd.Flags().GetString("type")
 
-			req := &metal.IPReservationRequestInput{
-				Metro:    &metro,
-				Tags:     tags,
-				Quantity: int32(quantity),
-				Type:     ttype,
-				Facility: &facility,
-			}
+			if typeFlag != "vrf" {
 
-			requestIPReservationRequest := &metal.RequestIPReservationRequest{
-				IPReservationRequestInput: req,
-			}
+				req = &metal.IPReservationRequestInput{
+					Metro:    &metro,
+					Tags:     tags,
+					Quantity: int32(quantity),
+					Type:     ttype,
+					Facility: &facility,
+				}
+
+				requestIPReservationRequest = &metal.RequestIPReservationRequest{
+					IPReservationRequestInput: req,
+				}
+			} else {
+				cidrFlag, _ := cmd.Flags().GetInt("cidr")
```

Comment:
`cidr`, `network`, and `vrfID` will already be set from https://github.com/equinix/metal-cli/pull/418/files?diff=split&w=1#diff-05b7b0cc0fa94c4a89c6c495544d266ff84a4325285bfbd67c7d936b963ff8d2R122-R126

Reply:
This does not seem to have been addressed.  The code is still reading the flag values instead of using the existing variables that were already hooked up to the flags in the lines Marques linked to.